<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="heinin" />

<meta name="date" content="2024-02-07" />

<title>Expression on Xenium base panel features in 13384 tumors</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GBM_Xenium</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Expression on Xenium base panel features in
13384 tumors</h1>
<h4 class="author">heinin</h4>
<h4 class="date">2024-02-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-02-08
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>GBM_Xenium/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240207code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240207)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240207code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240207)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/home/hnatri/GBM_Xenium/code/colors_themes.R
</td>
<td style="text-align:left;">
code/colors_themes.R
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/hnatri/GBM_Xenium/code/plot_functions.R
</td>
<td style="text-align:left;">
code/plot_functions.R
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/hnatri/GBM_Xenium/code/utilities.R
</td>
<td style="text-align:left;">
code/utilities.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsheininghpyotqDPOzg1riAN3RlPFomilCgqCKQW3U85AHgithubcomheininGBMXeniumtreef8fdf4b68098b7838d5af6aca4c38cb1a900582btargetblankf8fdf4ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://heinin: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@github.com/heinin/GBM_Xenium/tree/f8fdf4b68098b7838d5af6aca4c38cb1a900582b" target="_blank">f8fdf4b</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsheininghpyotqDPOzg1riAN3RlPFomilCgqCKQW3U85AHgithubcomheininGBMXeniumtreef8fdf4b68098b7838d5af6aca4c38cb1a900582btargetblankf8fdf4ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://heinin: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@github.com/heinin/GBM_Xenium/tree/f8fdf4b68098b7838d5af6aca4c38cb1a900582b" target="_blank">f8fdf4b</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   analysis/panel_feature_sc_expression.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/panel_feature_sc_expression.Rmd</code>) and HTML
(<code>docs/panel_feature_sc_expression.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://heinin: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@github.com/heinin/GBM_Xenium/blob/f8fdf4b68098b7838d5af6aca4c38cb1a900582b/docs/panel_feature_sc_expression.html" target="_blank">f8fdf4b</a>
</td>
<td>
heinin
</td>
<td>
2024-02-09
</td>
<td>
Updated the panel script
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://heinin: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@github.com/heinin/GBM_Xenium/blob/e962db5610852b6cc55c1415c849a837f7442476/analysis/panel_feature_sc_expression.Rmd" target="_blank">e962db5</a>
</td>
<td>
heinin
</td>
<td>
2024-02-08
</td>
<td>
Updated the panel script
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://heinin: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@github.com/heinin/GBM_Xenium/blob/e962db5610852b6cc55c1415c849a837f7442476/docs/panel_feature_sc_expression.html" target="_blank">e962db5</a>
</td>
<td>
heinin
</td>
<td>
2024-02-08
</td>
<td>
Updated the panel script
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://heinin: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@github.com/heinin/GBM_Xenium/blob/2d87bed53324a0c13a704c52e73eb35dec2359eb/analysis/panel_feature_sc_expression.Rmd" target="_blank">2d87bed</a>
</td>
<td>
heinin
</td>
<td>
2024-02-08
</td>
<td>
first commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://heinin: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@github.com/heinin/GBM_Xenium/blob/2d87bed53324a0c13a704c52e73eb35dec2359eb/docs/panel_feature_sc_expression.html" target="_blank">2d87bed</a>
</td>
<td>
heinin
</td>
<td>
2024-02-08
</td>
<td>
first commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="packages-and-environment-variables" class="section level3">
<h3>Packages and environment variables</h3>
<pre class="r"><code>suppressPackageStartupMessages({
  library(cli)
  library(Seurat)
  library(SeuratObject)
  library(SeuratDisk)
  library(tidyverse)
  library(tibble)
  library(ggplot2)
  library(ggpubr)
  library(ggrepel)
  library(workflowr)
  library(googlesheets4)
  library(UpSetR)})

setwd(&quot;/home/hnatri/ILD_ASE_Xenium/&quot;)
set.seed(9999)
options(ggrepel.max.overlaps = Inf)

# Colors, themes, cell type markers, and plot functions
source(&quot;/home/hnatri/GBM_Xenium/code/colors_themes.R&quot;)</code></pre>
<pre><code>✔ Reading from &quot;13384 tumor scRNAseq tables&quot;.</code></pre>
<pre><code>✔ Range &#39;&#39;Cluster annotations&#39;&#39;.</code></pre>
<pre><code>✔ Reading from &quot;13384 tumor scRNAseq tables&quot;.</code></pre>
<pre><code>✔ Range &#39;&#39;Cluster annotations, immune+fibroblast&#39;&#39;.</code></pre>
<pre class="r"><code>source(&quot;/home/hnatri/GBM_Xenium/code/plot_functions.R&quot;)
source(&quot;/home/hnatri/GBM_Xenium/code/utilities.R&quot;)</code></pre>
</div>
<div id="panel-features" class="section level3">
<h3>Panel features</h3>
<pre class="r"><code># https://docs.google.com/spreadsheets/d/14SQGru_Wj1AzzKhRb-ckivscl55xrr5Jw1hyjQim6zs/edit?usp=sharing
gs4_deauth()
panel_sheets  &lt;- gs4_get(&quot;https://docs.google.com/spreadsheets/d/14SQGru_Wj1AzzKhRb-ckivscl55xrr5Jw1hyjQim6zs/edit?usp=sharing&quot;)
sheet_names(panel_sheets)</code></pre>
<pre><code>[1] &quot;Info&quot;                                   
[2] &quot;Additional genes&quot;                       
[3] &quot;Multi-tissue/Cancer base panel&quot;         
[4] &quot;Brain base panel&quot;                       
[5] &quot;Breast cancer base panel&quot;               
[6] &quot;Berens lab GBM panel&quot;                   
[7] &quot;Multi-tissue/Cancer base panel + custom&quot;</code></pre>
<pre class="r"><code>breast &lt;- read_sheet(panel_sheets, sheet = &quot;Breast cancer base panel&quot;)</code></pre>
<pre><code>✔ Reading from &quot;GBM spatial transcriptomics&quot;.</code></pre>
<pre><code>✔ Range &#39;&#39;Breast cancer base panel&#39;&#39;.</code></pre>
<pre class="r"><code>brain &lt;- read_sheet(panel_sheets, sheet = &quot;Brain base panel&quot;)</code></pre>
<pre><code>✔ Reading from &quot;GBM spatial transcriptomics&quot;.</code></pre>
<pre><code>✔ Range &#39;&#39;Brain base panel&#39;&#39;.</code></pre>
<pre class="r"><code>multitissue_cancer &lt;- read_sheet(panel_sheets, sheet = &quot;Multi-tissue/Cancer base panel&quot;)</code></pre>
<pre><code>✔ Reading from &quot;GBM spatial transcriptomics&quot;.</code></pre>
<pre><code>✔ Range &#39;&#39;Multi-tissue/Cancer base panel&#39;&#39;.</code></pre>
<pre class="r"><code>gbm &lt;- read_sheet(panel_sheets, sheet = &quot;Berens lab GBM panel&quot;)</code></pre>
<pre><code>✔ Reading from &quot;GBM spatial transcriptomics&quot;.</code></pre>
<pre><code>✔ Range &#39;&#39;Berens lab GBM panel&#39;&#39;.</code></pre>
<pre class="r"><code>custom &lt;- read_sheet(panel_sheets, sheet = &quot;Additional genes&quot;)</code></pre>
<pre><code>✔ Reading from &quot;GBM spatial transcriptomics&quot;.</code></pre>
<pre><code>✔ Range &#39;&#39;Additional genes&#39;&#39;.</code></pre>
<pre class="r"><code>panel_gene_list &lt;- list(&quot;breast&quot; = breast$Name,
                        &quot;brain&quot; = brain$Gene,
                        &quot;multitissue_cancer&quot; = multitissue_cancer$Gene,
                        &quot;gbm&quot; = gbm$Name,
                        &quot;custom&quot; = custom$Gene)

panel_gene_list &lt;- panel_gene_list[order(sapply(panel_gene_list, length))]

# Cell type annotations for genes in the brain panel
brain$`Annotation (Xenium)` %&gt;% table() %&gt;%
  as.data.frame() %&gt;% 
  dplyr::rename(celltype = &quot;.&quot;) %&gt;%
  ggplot(aes(x = reorder(celltype, -Freq), y = Freq)) +
    geom_bar(stat = &quot;identity&quot;)+
    theme_bw() +
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
    xlab(&quot;Cell type&quot;) +
    ylab(&quot;# features&quot;)</code></pre>
<p><img src="figure/panel_feature_sc_expression.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://heinin: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@github.com/heinin/GBM_Xenium/blob/f8fdf4b68098b7838d5af6aca4c38cb1a900582b/docs/figure/panel_feature_sc_expression.Rmd/unnamed-chunk-2-1.png" target="_blank">f8fdf4b</a>
</td>
<td>
heinin
</td>
<td>
2024-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="import-data" class="section level3">
<h3>Import data</h3>
<pre class="r"><code>seurat_object &lt;- readRDS(&quot;/tgen_labs/banovich/BCTCSF/Heini/tumor_integrated_UPN109pre_noUPN208_soupX_snn_metadata_no4_7.rds&quot;)
DefaultAssay(seurat_object)</code></pre>
<pre><code>[1] &quot;RNA&quot;</code></pre>
<pre class="r"><code># UMAP with cell type annotations
DimPlot(seurat_object,
        group.by = &quot;celltype&quot;,
        cols = tumor_celltype_col,
        reduction = &quot;integrated_sct_umap&quot;,
        label = T,
        label.box = T,
        label.size = 3,
        repel = T,
        raster = T,
        raster.dpi = c(1024, 1024),
        pt.size = 3) +
  ggtitle(&quot;&quot;) +
  theme_classic() +
  manuscript_theme + 
  NoLegend() +
  NoAxes() +
  coord_fixed(1)</code></pre>
<p><img src="figure/panel_feature_sc_expression.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://heinin: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@github.com/heinin/GBM_Xenium/blob/f8fdf4b68098b7838d5af6aca4c38cb1a900582b/docs/figure/panel_feature_sc_expression.Rmd/unnamed-chunk-3-1.png" target="_blank">f8fdf4b</a>
</td>
<td>
heinin
</td>
<td>
2024-02-09
</td>
</tr>
<tr>
<td>
<a href="https://heinin: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@github.com/heinin/GBM_Xenium/blob/2d87bed53324a0c13a704c52e73eb35dec2359eb/docs/figure/panel_feature_sc_expression.Rmd/unnamed-chunk-3-1.png" target="_blank">2d87bed</a>
</td>
<td>
heinin
</td>
<td>
2024-02-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Pseudobulk expression (CPM) across cell types
exp_data &lt;- AggregateExpression(seurat_object,
                                assay = &quot;RNA&quot;,
                                group.by = &quot;celltype&quot;,
                                normalization.method = &quot;LogNormalize&quot;,
                                scale.factor = 1e6)
exp_data &lt;- as.matrix(exp_data$RNA)

# Porportion of cells expression and (scaled) average expression
p &lt;- DotPlot(object = seurat_object,
             features = unique(unlist(panel_gene_list)),
             group.by = &quot;celltype&quot;)</code></pre>
<pre><code>Warning: Could not find CD11b in the default search locations, found in
&#39;Protein&#39; assay instead</code></pre>
<pre><code>Warning: Could not find CD11c in the default search locations, found in
&#39;Protein&#39; assay instead</code></pre>
<pre><code>Warning: The following requested variables were not found: CCL9, FBLN4, IFNGR,
IL12, opalin, sox4, tek, TNFR1</code></pre>
<pre class="r"><code># Extracting values
df &lt;- p$data

# Removing NaN values
if(nrow(df[which(is.nan(df$avg.exp.scaled)),])&gt;0){
  df &lt;- df[-which(is.nan(df$avg.exp.scaled)),]
}

# Which panel features are present in our annotations?
lapply(panel_gene_list, function(xx){
  (length(intersect(rownames(exp_data), xx))/length(xx))*100
})</code></pre>
<pre><code>$custom
[1] 89.71963

$gbm
[1] 100

$brain
[1] 100

$breast
[1] 100

$multitissue_cancer
[1] 100</code></pre>
</div>
<div id="overlap-of-panel-features" class="section level3">
<h3>Overlap of panel features</h3>
<pre class="r"><code>upset(fromList(panel_gene_list), order.by = &quot;freq&quot;)</code></pre>
<p><img src="figure/panel_feature_sc_expression.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://heinin: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@github.com/heinin/GBM_Xenium/blob/2d87bed53324a0c13a704c52e73eb35dec2359eb/docs/figure/panel_feature_sc_expression.Rmd/unnamed-chunk-4-1.png" target="_blank">2d87bed</a>
</td>
<td>
heinin
</td>
<td>
2024-02-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-feature-expression-across-cell-types"
class="section level3">
<h3>Panel feature expression across cell types</h3>
<pre class="r"><code># A list of genes meeting the expression threshold
exp_list &lt;- lapply(unique(df$id), function(xx){
  
  df %&gt;% filter(id == xx,
                pct.exp &gt; 0.05,
                avg.exp &gt; 1) %&gt;%
    dplyr::select(features.plot) %&gt;%
    unlist() %&gt;% as.character()
})
names(exp_list) &lt;- unique(df$id)

exp_list &lt;- exp_list[order(sapply(exp_list, length))]</code></pre>
</div>
<div
id="panel-features-expressed-in-scrnaseq-with-5-of-cells-expressing-avg.-exp-1"
class="section level3">
<h3>Panel features expressed in scRNAseq with &gt;5% of cells
expressing, avg. exp &gt;1</h3>
<pre class="r"><code>upset(fromList(c(panel_gene_list, exp_list)),
      keep.order = TRUE,
      order.by = &quot;freq&quot;,
      nsets = length(c(panel_gene_list, exp_list)),
      mb.ratio = c(0.4, 0.6))</code></pre>
<p><img src="figure/panel_feature_sc_expression.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://heinin: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@github.com/heinin/GBM_Xenium/blob/f8fdf4b68098b7838d5af6aca4c38cb1a900582b/docs/figure/panel_feature_sc_expression.Rmd/unnamed-chunk-6-1.png" target="_blank">f8fdf4b</a>
</td>
<td>
heinin
</td>
<td>
2024-02-09
</td>
</tr>
<tr>
<td>
<a href="https://heinin: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@github.com/heinin/GBM_Xenium/blob/2d87bed53324a0c13a704c52e73eb35dec2359eb/docs/figure/panel_feature_sc_expression.Rmd/unnamed-chunk-6-1.png" target="_blank">2d87bed</a>
</td>
<td>
heinin
</td>
<td>
2024-02-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>bar_cols &lt;- c(rep(&quot;deeppink3&quot;, length(panel_gene_list)),
              rep(&quot;aquamarine3&quot;, length(exp_list)))
names(bar_cols) &lt;- c(names(panel_gene_list), names(exp_list))

upset(fromList(c(panel_gene_list, exp_list)),
      keep.order = TRUE,
      sets = c(names(panel_gene_list), names(exp_list)),
      order.by = &quot;freq&quot;,
      nsets = length(c(panel_gene_list, exp_list)),
      nintersects = 60,
      sets.bar.color = bar_cols,
      mb.ratio = c(0.4, 0.6))</code></pre>
<p><img src="figure/panel_feature_sc_expression.Rmd/unnamed-chunk-6-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-2">
Past versions of unnamed-chunk-6-2.png
</button>
</p>
<div id="fig-unnamed-chunk-6-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://heinin: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@github.com/heinin/GBM_Xenium/blob/f8fdf4b68098b7838d5af6aca4c38cb1a900582b/docs/figure/panel_feature_sc_expression.Rmd/unnamed-chunk-6-2.png" target="_blank">f8fdf4b</a>
</td>
<td>
heinin
</td>
<td>
2024-02-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="same-with-1-expressing-avg.-exp-1" class="section level3">
<h3>Same with &gt;1% expressing, avg. exp &gt;1</h3>
<pre class="r"><code>exp_list &lt;- lapply(unique(df$id), function(xx){
  
  df %&gt;% filter(id == xx,
                pct.exp &gt; 0.01,
                avg.exp &gt; 1) %&gt;%
    dplyr::select(features.plot) %&gt;%
    unlist() %&gt;% as.character()
})
names(exp_list) &lt;- unique(df$id)

exp_list &lt;- exp_list[order(sapply(exp_list, length))]</code></pre>
<pre class="r"><code>bar_cols &lt;- c(rep(&quot;deeppink3&quot;, length(panel_gene_list)),
              rep(&quot;aquamarine3&quot;, length(exp_list)))
names(bar_cols) &lt;- c(names(panel_gene_list), names(exp_list))

upset(fromList(c(panel_gene_list, exp_list)),
      keep.order = TRUE,
      sets = c(names(panel_gene_list), names(exp_list)),
      order.by = &quot;freq&quot;,
      nsets = length(c(panel_gene_list, exp_list)),
      nintersects = 50,
      sets.bar.color = bar_cols,
      mb.ratio = c(0.4, 0.6))</code></pre>
<p><img src="figure/panel_feature_sc_expression.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://heinin: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@github.com/heinin/GBM_Xenium/blob/f8fdf4b68098b7838d5af6aca4c38cb1a900582b/docs/figure/panel_feature_sc_expression.Rmd/unnamed-chunk-8-1.png" target="_blank">f8fdf4b</a>
</td>
<td>
heinin
</td>
<td>
2024-02-09
</td>
</tr>
<tr>
<td>
<a href="https://heinin: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@github.com/heinin/GBM_Xenium/blob/e962db5610852b6cc55c1415c849a837f7442476/docs/figure/panel_feature_sc_expression.Rmd/unnamed-chunk-8-1.png" target="_blank">e962db5</a>
</td>
<td>
heinin
</td>
<td>
2024-02-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="looking-at-genes-in-intersects" class="section level3">
<h3>Looking at genes in intersects</h3>
<div
id="genes-in-the-barens-lab-gbm-panel-and-expressed-in-any-cell-type"
class="section level4">
<h4>Genes in the Barens lab GBM panel and expressed in any cell
type</h4>
<pre class="r"><code># Intersect
intersect(panel_gene_list[[&quot;gbm&quot;]], unlist(exp_list))</code></pre>
<pre><code> [1] &quot;CD4&quot;      &quot;ACTN1&quot;    &quot;SERPINE1&quot; &quot;RPS6KB1&quot;  &quot;CCND1&quot;    &quot;CDKN1B&quot;  
 [7] &quot;CDKN1A&quot;   &quot;SOX9&quot;     &quot;EGLN3&quot;    &quot;PDGFRA&quot;   &quot;THBS1&quot;    &quot;LMNB1&quot;   
[13] &quot;CD44&quot;     &quot;EPAS1&quot;    &quot;PIK3CA&quot;   &quot;MAP1LC3B&quot; &quot;SLC20A1&quot;  &quot;THY1&quot;    
[19] &quot;CSF1R&quot;    &quot;ITGA5&quot;    &quot;LAPTM5&quot;   &quot;HHIP&quot;     &quot;COL1A2&quot;   &quot;TPP1&quot;    
[25] &quot;IL7R&quot;     &quot;ITGA3&quot;    &quot;ZEB1&quot;     &quot;RAB8B&quot;    &quot;SOX2&quot;     &quot;HIF1A&quot;   
[31] &quot;P2RX4&quot;    &quot;ARHGEF3&quot;  &quot;UBC&quot;      &quot;PLAUR&quot;    &quot;CD8A&quot;     &quot;IL1RN&quot;   
[37] &quot;ATG7&quot;     &quot;DLL3&quot;     &quot;SOX10&quot;    &quot;PTGS2&quot;    &quot;NES&quot;      &quot;NKX6-2&quot;  
[43] &quot;MKI67&quot;    &quot;ADGRL4&quot;   &quot;VEGFA&quot;    &quot;NRP1&quot;     &quot;IL2RA&quot;    &quot;BNIP3L&quot;  
[49] &quot;IGFBP3&quot;   &quot;GPNMB&quot;    &quot;OLIG2&quot;    &quot;STAT4&quot;    &quot;CNP&quot;      &quot;UBE2D3&quot;  
[55] &quot;CD72&quot;     &quot;RNH1&quot;     &quot;CX3CR1&quot;   &quot;AKAP12&quot;   &quot;ZEB2&quot;     &quot;DCN&quot;     
[61] &quot;ENPP2&quot;    &quot;IL1R2&quot;    &quot;IFI44&quot;    &quot;GPR17&quot;    &quot;TGFBI&quot;    &quot;FYB1&quot;    
[67] &quot;BMPR1B&quot;   &quot;PTP4A3&quot;   &quot;CD84&quot;     &quot;AQP4&quot;     &quot;TREM1&quot;    &quot;OPALIN&quot;  
[73] &quot;FOXP1&quot;    &quot;FLT1&quot;     &quot;CLDN5&quot;    &quot;PLP1&quot;     &quot;BCL2L11&quot;  &quot;BID&quot;     
[79] &quot;ATG5&quot;     &quot;GFAP&quot;     &quot;SPP1&quot;    </code></pre>
<pre class="r"><code># Panel genes not expressed in scRNAseq
setdiff(panel_gene_list[[&quot;gbm&quot;]], unlist(exp_list))</code></pre>
<pre><code> [1] &quot;SNAI2&quot;    &quot;CDC6&quot;     &quot;SALL4&quot;    &quot;SLC32A1&quot;  &quot;STMN2&quot;    &quot;SLC17A7&quot; 
 [7] &quot;NANOG&quot;    &quot;MMP13&quot;    &quot;SYP&quot;      &quot;CXCL9&quot;    &quot;CDK2&quot;     &quot;CDKN2B&quot;  
[13] &quot;VCAM1&quot;    &quot;FASTK&quot;    &quot;ALDH1A1&quot;  &quot;DCHS1&quot;    &quot;PLK1&quot;     &quot;SLAMF1&quot;  
[19] &quot;RRM2&quot;     &quot;PTGDR&quot;    &quot;GLB1&quot;     &quot;GPR171&quot;   &quot;TERT&quot;     &quot;GATD1&quot;   
[25] &quot;PPIE&quot;     &quot;TMEM130&quot;  &quot;CLEC14A&quot;  &quot;FZD9&quot;     &quot;CD34&quot;     &quot;L1CAM&quot;   
[31] &quot;MOXD1&quot;    &quot;BAK1&quot;     &quot;FOXP3&quot;    &quot;GAD2&quot;     &quot;SUV39H1&quot;  &quot;TEK&quot;     
[37] &quot;TBR1&quot;     &quot;TMEM119&quot;  &quot;SMCR8&quot;    &quot;CDCA7L&quot;   &quot;ACP5&quot;     &quot;KBTBD8&quot;  
[43] &quot;BRAP&quot;     &quot;TOP2A&quot;    &quot;MLH1&quot;     &quot;AQP1&quot;     &quot;CDK1&quot;     &quot;DNASE1L3&quot;
[49] &quot;ATG12&quot;    &quot;ADAMDEC1&quot; &quot;SMPD1&quot;    &quot;PROM1&quot;    &quot;SPON1&quot;    &quot;BECN1&quot;   
[55] &quot;MYSM1&quot;   </code></pre>
<pre class="r"><code>#gbm %&gt;% filter(Name %in% setdiff(panel_gene_list[[&quot;gbm&quot;]], unlist(exp_list)))

# Total # of panel features expressed in scRNAseq
length(intersect(panel_gene_list[[&quot;gbm&quot;]], unique(unlist(exp_list))))</code></pre>
<pre><code>[1] 81</code></pre>
<pre class="r"><code># % out of the whole panel expressed in scRNAseq
length(intersect(panel_gene_list[[&quot;gbm&quot;]], unique(unlist(exp_list))))/length(panel_gene_list[[&quot;gbm&quot;]])*100</code></pre>
<pre><code>[1] 59.55882</code></pre>
</div>
<div
id="genes-in-the-multi-tissuecancer-panel-and-expressed-in-any-cell-type"
class="section level4">
<h4>Genes in the multi-tissue/cancer panel and expressed in any cell
type</h4>
<pre class="r"><code># Intersect
intersect(panel_gene_list[[&quot;multitissue_cancer&quot;]], unlist(exp_list))</code></pre>
<pre><code>  [1] &quot;ACTA2&quot;    &quot;ACTG2&quot;    &quot;ADAM28&quot;   &quot;ADAMTS1&quot;  &quot;ADGRL4&quot;   &quot;AIF1&quot;    
  [7] &quot;ANGPT2&quot;   &quot;ANPEP&quot;    &quot;APOBEC3A&quot; &quot;APOLD1&quot;   &quot;AQP9&quot;     &quot;ARFGEF3&quot; 
 [13] &quot;BAMBI&quot;    &quot;BANK1&quot;    &quot;BASP1&quot;    &quot;BCL2L11&quot;  &quot;C15orf48&quot; &quot;C1orf162&quot;
 [19] &quot;C7&quot;       &quot;CAV1&quot;     &quot;CAVIN1&quot;   &quot;CCL5&quot;     &quot;CCR7&quot;     &quot;CD14&quot;    
 [25] &quot;CD163&quot;    &quot;CD1A&quot;     &quot;CD1C&quot;     &quot;CD2&quot;      &quot;CD247&quot;    &quot;CD27&quot;    
 [31] &quot;CD28&quot;     &quot;CD300E&quot;   &quot;CD3D&quot;     &quot;CD3E&quot;     &quot;CD4&quot;      &quot;CD68&quot;    
 [37] &quot;CD69&quot;     &quot;CD79A&quot;    &quot;CD83&quot;     &quot;CD86&quot;     &quot;CD8A&quot;     &quot;CD93&quot;    
 [43] &quot;CENPF&quot;    &quot;CLEC10A&quot;  &quot;CLEC4E&quot;   &quot;COL5A2&quot;   &quot;CRISPLD2&quot; &quot;CSF2RA&quot;  
 [49] &quot;CTLA4&quot;    &quot;CXCL10&quot;   &quot;CXCL2&quot;    &quot;CXCR4&quot;    &quot;CYTIP&quot;    &quot;DERL3&quot;   
 [55] &quot;DNAAF1&quot;   &quot;DST&quot;      &quot;DUSP2&quot;    &quot;EGFL7&quot;    &quot;EGFR&quot;     &quot;FAS&quot;     
 [61] &quot;FBLN1&quot;    &quot;FBN1&quot;     &quot;FCER1A&quot;   &quot;FCGR1A&quot;   &quot;FCGR3A&quot;   &quot;FCN1&quot;    
 [67] &quot;FGFBP2&quot;   &quot;FGL2&quot;     &quot;FHL2&quot;     &quot;FKBP11&quot;   &quot;GATM&quot;     &quot;GDF15&quot;   
 [73] &quot;GEM&quot;      &quot;GLIPR1&quot;   &quot;GNG11&quot;    &quot;GNLY&quot;     &quot;GPC1&quot;     &quot;GPR183&quot;  
 [79] &quot;GPRC5A&quot;   &quot;GZMA&quot;     &quot;GZMB&quot;     &quot;GZMK&quot;     &quot;HAMP&quot;     &quot;HAVCR2&quot;  
 [85] &quot;HES4&quot;     &quot;HIGD1B&quot;   &quot;HLA-DQB2&quot; &quot;IGSF6&quot;    &quot;IL1R2&quot;    &quot;IL2RA&quot;   
 [91] &quot;IL3RA&quot;    &quot;IL7R&quot;     &quot;IRF8&quot;     &quot;KCNMA1&quot;   &quot;KLRB1&quot;    &quot;KLRD1&quot;   
 [97] &quot;LAG3&quot;     &quot;LILRA4&quot;   &quot;LILRB2&quot;   &quot;LILRB4&quot;   &quot;LPL&quot;      &quot;LY86&quot;    
[103] &quot;MARCO&quot;    &quot;MCF2L&quot;    &quot;MDM2&quot;     &quot;MEF2C&quot;    &quot;MEST&quot;     &quot;MET&quot;     
[109] &quot;MKI67&quot;    &quot;MNDA&quot;     &quot;MS4A1&quot;    &quot;MS4A4A&quot;   &quot;MS4A6A&quot;   &quot;MYC&quot;     
[115] &quot;MYLK&quot;     &quot;MZB1&quot;     &quot;NKG7&quot;     &quot;NPDC1&quot;    &quot;PCOLCE&quot;   &quot;PDGFRA&quot;  
[121] &quot;PDGFRB&quot;   &quot;PDPN&quot;     &quot;PECAM1&quot;   &quot;PLAC9&quot;    &quot;PLCG2&quot;    &quot;PLD4&quot;    
[127] &quot;PMP22&quot;    &quot;PPARG&quot;    &quot;PRDM1&quot;    &quot;PRF1&quot;     &quot;PROX1&quot;    &quot;PTGDS&quot;   
[133] &quot;PTN&quot;      &quot;PTPRC&quot;    &quot;RAMP2&quot;    &quot;RETN&quot;     &quot;RGS16&quot;    &quot;S100A1&quot;  
[139] &quot;S100A12&quot;  &quot;SELL&quot;     &quot;SEMA3C&quot;   &quot;SERPINB2&quot; &quot;SERPINB9&quot; &quot;SFRP4&quot;   
[145] &quot;SOX2&quot;     &quot;SPI1&quot;     &quot;SPIB&quot;     &quot;SRPX&quot;     &quot;TCF4&quot;     &quot;TCIM&quot;    
[151] &quot;TCL1A&quot;    &quot;TENT5C&quot;   &quot;TFPI&quot;     &quot;THY1&quot;     &quot;TNC&quot;      &quot;TNFRSF9&quot; 
[157] &quot;TREM2&quot;    &quot;UBE2C&quot;    &quot;VCAN&quot;     &quot;VSIG4&quot;    &quot;VWF&quot;     </code></pre>
<pre class="r"><code># Panel genes not expressed in scRNAseq
setdiff(panel_gene_list[[&quot;multitissue_cancer&quot;]], unlist(exp_list))</code></pre>
<pre><code>  [1] &quot;ABCC11&quot;    &quot;ACE2&quot;      &quot;ACKR1&quot;     &quot;ADGRE1&quot;    &quot;ADH1C&quot;     &quot;ADH4&quot;     
  [7] &quot;ADIPOQ&quot;    &quot;AGER&quot;      &quot;AGR3&quot;      &quot;AHSP&quot;      &quot;ALAS2&quot;     &quot;ALDH1A3&quot;  
 [13] &quot;AMY2A&quot;     &quot;APCDD1&quot;    &quot;APOA5&quot;     &quot;AQP2&quot;      &quot;AQP3&quot;      &quot;AQP8&quot;     
 [19] &quot;AR&quot;        &quot;ASCL1&quot;     &quot;ASCL3&quot;     &quot;ASPN&quot;      &quot;BBOX1&quot;     &quot;BMX&quot;      
 [25] &quot;BTNL9&quot;     &quot;C1orf194&quot;  &quot;C20orf85&quot;  &quot;C5orf46&quot;   &quot;C6orf118&quot;  &quot;CA4&quot;      
 [31] &quot;CAPN8&quot;     &quot;CAVIN2&quot;    &quot;CCDC39&quot;    &quot;CCDC78&quot;    &quot;CCL19&quot;     &quot;CCL27&quot;    
 [37] &quot;CCNB2&quot;     &quot;CCR2&quot;      &quot;CD19&quot;      &quot;CD1E&quot;      &quot;CD274&quot;     &quot;CD34&quot;     
 [43] &quot;CD5L&quot;      &quot;CD70&quot;      &quot;CDH16&quot;     &quot;CDK1&quot;      &quot;CFAP53&quot;    &quot;CFB&quot;      
 [49] &quot;CFHR1&quot;     &quot;CFHR3&quot;     &quot;CFTR&quot;      &quot;CHGA&quot;      &quot;CLCA1&quot;     &quot;CLCA2&quot;    
 [55] &quot;CLEC14A&quot;   &quot;CLECL1&quot;    &quot;CLIC6&quot;     &quot;CNN1&quot;      &quot;COCH&quot;      &quot;COL17A1&quot;  
 [61] &quot;CPA3&quot;      &quot;CRHBP&quot;     &quot;CSF3&quot;      &quot;CTSG&quot;      &quot;CTSK&quot;      &quot;CXCL6&quot;    
 [67] &quot;CXCL9&quot;     &quot;CYP1A1&quot;    &quot;CYP2A7&quot;    &quot;CYP2B6&quot;    &quot;CYP2F1&quot;    &quot;CYP3A4&quot;   
 [73] &quot;CYP4B1&quot;    &quot;DES&quot;       &quot;DIRAS3&quot;    &quot;DMBT1&quot;     &quot;DNASE1L3&quot;  &quot;DPEP1&quot;    
 [79] &quot;DPT&quot;       &quot;ECSCR&quot;     &quot;EDN1&quot;      &quot;EDNRB&quot;     &quot;EHF&quot;       &quot;ELF5&quot;     
 [85] &quot;EPCAM&quot;     &quot;ERBB2&quot;     &quot;ERG&quot;       &quot;ESR1&quot;      &quot;FCN2&quot;      &quot;FGFBP1&quot;   
 [91] &quot;FOXA1&quot;     &quot;FOXI1&quot;     &quot;FOXJ1&quot;     &quot;FOXP3&quot;     &quot;FSTL3&quot;     &quot;FXYD2&quot;    
 [97] &quot;GATA2&quot;     &quot;GCG&quot;       &quot;GHRL&quot;      &quot;GKN2&quot;      &quot;GLYATL1&quot;   &quot;GPC3&quot;     
[103] &quot;GPX2&quot;      &quot;GYPA&quot;      &quot;GYPB&quot;      &quot;HEMGN&quot;     &quot;HEPACAM2&quot;  &quot;HMGCS2&quot;   
[109] &quot;HPGDS&quot;     &quot;HPX&quot;       &quot;IGF1&quot;      &quot;IL1RL1&quot;    &quot;INMT&quot;      &quot;INS&quot;      
[115] &quot;KCNK3&quot;     &quot;KIT&quot;       &quot;KLK11&quot;     &quot;KLRC1&quot;     &quot;KNG1&quot;      &quot;KRT20&quot;    
[121] &quot;KRT7&quot;      &quot;LAMP3&quot;     &quot;LGI4&quot;      &quot;LGR5&quot;      &quot;LIF&quot;       &quot;LILRA5&quot;   
[127] &quot;LTBP2&quot;     &quot;LY6D&quot;      &quot;LYVE1&quot;     &quot;MALL&quot;      &quot;MAMDC2&quot;    &quot;MCEMP1&quot;   
[133] &quot;MEDAG&quot;     &quot;MFAP5&quot;     &quot;MLANA&quot;     &quot;MLPH&quot;      &quot;MMRN1&quot;     &quot;MMRN2&quot;    
[139] &quot;MPEG1&quot;     &quot;MRC1&quot;      &quot;MS4A2&quot;     &quot;MTRNR2L11&quot; &quot;MYBPC1&quot;    &quot;MYH11&quot;    
[145] &quot;NAT8&quot;      &quot;NTN4&quot;      &quot;OGN&quot;       &quot;OPRPN&quot;     &quot;PCNA&quot;      &quot;PCP4&quot;     
[151] &quot;PCSK2&quot;     &quot;PDCD1&quot;     &quot;PEBP4&quot;     &quot;PGR&quot;       &quot;PLA2G7&quot;    &quot;PLIN4&quot;    
[157] &quot;PPP1R1A&quot;   &quot;PPP1R1B&quot;   &quot;PPY&quot;       &quot;PRG4&quot;      &quot;PVALB&quot;     &quot;RAPGEF3&quot;  
[163] &quot;RBP5&quot;      &quot;RERGL&quot;     &quot;RIDA&quot;      &quot;RND1&quot;      &quot;RTKN2&quot;     &quot;SCGB2A1&quot;  
[169] &quot;SCGN&quot;      &quot;SELE&quot;      &quot;SERPINB3&quot;  &quot;SFRP2&quot;     &quot;SFTA2&quot;     &quot;SH2D3C&quot;   
[175] &quot;SLAMF1&quot;    &quot;SLAMF7&quot;    &quot;SLC18A2&quot;   &quot;SLC22A8&quot;   &quot;SLC26A2&quot;   &quot;SLC26A3&quot;  
[181] &quot;SLC4A1&quot;    &quot;SMIM24&quot;    &quot;SMYD2&quot;     &quot;SNAI1&quot;     &quot;SNCA&quot;      &quot;SNCG&quot;     
[187] &quot;SNTN&quot;      &quot;SOX17&quot;     &quot;SOX18&quot;     &quot;SPDEF&quot;     &quot;SST&quot;       &quot;STC1&quot;     
[193] &quot;STC2&quot;      &quot;STEAP4&quot;    &quot;TAC1&quot;      &quot;TAT&quot;       &quot;TBX3&quot;      &quot;TCF15&quot;    
[199] &quot;TFF2&quot;      &quot;THAP2&quot;     &quot;THBS2&quot;     &quot;TIMP4&quot;     &quot;TM4SF18&quot;   &quot;TM4SF4&quot;   
[205] &quot;TMC5&quot;      &quot;TMEM100&quot;   &quot;TMEM174&quot;   &quot;TMEM52B&quot;   &quot;TNFRSF13B&quot; &quot;TNFRSF17&quot; 
[211] &quot;TOP2A&quot;     &quot;TRAC&quot;      &quot;TSPAN19&quot;   &quot;UMOD&quot;      &quot;UPK3B&quot;     &quot;VWA5A&quot;    </code></pre>
<pre class="r"><code># Annotations for genes not expressed in scRNAseq
multitissue_cancer %&gt;% filter(!Gene %in% unlist(exp_list))</code></pre>
<pre><code># A tibble: 216 × 3
   Gene   Ensembl_ID      Annotation                                            
   &lt;chr&gt;  &lt;chr&gt;           &lt;chr&gt;                                                 
 1 ABCC11 ENSG00000121270 Breast - breast glandular cells                       
 2 ACE2   ENSG00000130234 General - Smooth muscle cells,Kidney - Proximal tubul…
 3 ACKR1  ENSG00000213088 General - Endothelial cells                           
 4 ADGRE1 ENSG00000174837 Liver - kupffer cells                                 
 5 ADH1C  ENSG00000248144 Colon - Undifferentiated cells,Liver - Hepatocytes,Pa…
 6 ADH4   ENSG00000198099 Liver - Hepatocytes                                   
 7 ADIPOQ ENSG00000181092 Breast - adipocytes                                   
 8 AGER   ENSG00000204305 Lung - Alveolar cells type 1                          
 9 AGR3   ENSG00000173467 Breast - Breast glandular cells,Colon - Distal entero…
10 AHSP   ENSG00000169877 Liver - Erythroid cells                               
# ℹ 206 more rows</code></pre>
<pre class="r"><code># Total # of panel features expressed in scRNAseq
length(intersect(panel_gene_list[[&quot;multitissue_cancer&quot;]], unique(unlist(exp_list))))</code></pre>
<pre><code>[1] 161</code></pre>
<pre class="r"><code># % out of the whole panel expressed in scRNAseq
length(intersect(panel_gene_list[[&quot;multitissue_cancer&quot;]], unique(unlist(exp_list))))/length(panel_gene_list[[&quot;multitissue_cancer&quot;]])*100</code></pre>
<pre><code>[1] 42.70557</code></pre>
</div>
<div id="genes-in-the-brain-panel-and-expressed-in-any-cell-type"
class="section level4">
<h4>Genes in the brain panel and expressed in any cell type</h4>
<pre class="r"><code># Intersect
intersect(panel_gene_list[[&quot;brain&quot;]], unlist(exp_list))</code></pre>
<pre><code>  [1] &quot;AIF1&quot;     &quot;ALK&quot;      &quot;ANXA1&quot;    &quot;APOE&quot;     &quot;APP&quot;      &quot;AQP4&quot;    
  [7] &quot;ARHGAP24&quot; &quot;BCAN&quot;     &quot;BRINP3&quot;   &quot;BTBD11&quot;   &quot;C1orf162&quot; &quot;CALCRL&quot;  
 [13] &quot;CAPG&quot;     &quot;CAPN3&quot;    &quot;CAV1&quot;     &quot;CCL4&quot;     &quot;CCL5&quot;     &quot;CD14&quot;    
 [19] &quot;CD163&quot;    &quot;CD2&quot;      &quot;CD3G&quot;     &quot;CD4&quot;      &quot;CD48&quot;     &quot;CD52&quot;    
 [25] &quot;CD68&quot;     &quot;CD83&quot;     &quot;CD86&quot;     &quot;CDH4&quot;     &quot;CDH6&quot;     &quot;CEMIP2&quot;  
 [31] &quot;CENPF&quot;    &quot;CLDN11&quot;   &quot;CNDP1&quot;    &quot;CNTN2&quot;    &quot;COL12A1&quot;  &quot;CORO1A&quot;  
 [37] &quot;CSPG4&quot;    &quot;CTNNA3&quot;   &quot;CTSH&quot;     &quot;CTSS&quot;     &quot;CX3CR1&quot;   &quot;CXCR4&quot;   
 [43] &quot;CYTIP&quot;    &quot;DCN&quot;      &quot;DNER&quot;     &quot;EFHD1&quot;    &quot;EGFR&quot;     &quot;ERBB3&quot;   
 [49] &quot;ERMN&quot;     &quot;EYA4&quot;     &quot;FBLN1&quot;    &quot;FCER1G&quot;   &quot;FCGR1A&quot;   &quot;FCGR3A&quot;  
 [55] &quot;FGFR2&quot;    &quot;FLT1&quot;     &quot;GNLY&quot;     &quot;GPNMB&quot;    &quot;GPR183&quot;   &quot;GPR34&quot;   
 [61] &quot;GZMA&quot;     &quot;HHATL&quot;    &quot;HILPDA&quot;   &quot;HLA-DMB&quot;  &quot;HLA-DQA1&quot; &quot;HS3ST4&quot;  
 [67] &quot;IDH1&quot;     &quot;IDH2&quot;     &quot;IDO1&quot;     &quot;IFITM3&quot;   &quot;IGFBP3&quot;   &quot;IGFBP4&quot;  
 [73] &quot;IGFBP5&quot;   &quot;IL7R&quot;     &quot;ITGAX&quot;    &quot;ITGB2&quot;    &quot;KLF2&quot;     &quot;KLF4&quot;    
 [79] &quot;KLK6&quot;     &quot;KLRB1&quot;    &quot;LAMA2&quot;    &quot;LOX&quot;      &quot;LY86&quot;     &quot;MAG&quot;     
 [85] &quot;MAL&quot;      &quot;MCTP2&quot;    &quot;MEIS2&quot;    &quot;MGST1&quot;    &quot;MKI67&quot;    &quot;MOBP&quot;    
 [91] &quot;MOG&quot;      &quot;MS4A6A&quot;   &quot;MYO16&quot;    &quot;MYRF&quot;     &quot;NES&quot;      &quot;NKG7&quot;    
 [97] &quot;NNAT&quot;     &quot;NR2F2&quot;    &quot;NR4A2&quot;    &quot;NRP1&quot;     &quot;OLIG1&quot;    &quot;OLIG2&quot;   
[103] &quot;OPALIN&quot;   &quot;PCSK6&quot;    &quot;PDGFRA&quot;   &quot;PECAM1&quot;   &quot;PLCE1&quot;    &quot;POSTN&quot;   
[109] &quot;PROX1&quot;    &quot;PSEN1&quot;    &quot;PSENEN&quot;   &quot;PTPRC&quot;    &quot;PTPRZ1&quot;   &quot;RASGRP1&quot; 
[115] &quot;RFTN1&quot;    &quot;RGS10&quot;    &quot;RGS16&quot;    &quot;RNASET2&quot;  &quot;RNF144B&quot;  &quot;RYR3&quot;    
[121] &quot;S100A4&quot;   &quot;SDK1&quot;     &quot;SEMA5A&quot;   &quot;SLC24A3&quot;  &quot;SLIT3&quot;    &quot;SNTB2&quot;   
[127] &quot;SOX10&quot;    &quot;SOX2&quot;     &quot;SOX4&quot;     &quot;SOX9&quot;     &quot;SPI1&quot;     &quot;ST18&quot;    
[133] &quot;STAT3&quot;    &quot;STK32B&quot;   &quot;STXBP2&quot;   &quot;TENM1&quot;    &quot;TGFB1&quot;    &quot;TGFB2&quot;   
[139] &quot;TGFBI&quot;    &quot;THBS1&quot;    &quot;THEMIS&quot;   &quot;TMEM132C&quot; &quot;TMIGD3&quot;   &quot;TREM2&quot;   
[145] &quot;TSHZ2&quot;    &quot;TTYH1&quot;    &quot;UGT8&quot;     &quot;UNC5B&quot;    &quot;VCAN&quot;    </code></pre>
<pre class="r"><code># Panel genes not expressed in scRNAseq
setdiff(panel_gene_list[[&quot;brain&quot;]], unlist(exp_list))</code></pre>
<pre><code>  [1] &quot;ABCC9&quot;    &quot;ADAMTS12&quot; &quot;ADAMTS16&quot; &quot;ADAMTS3&quot;  &quot;ADRA1A&quot;   &quot;ADRA1B&quot;  
  [7] &quot;ANGPT1&quot;   &quot;ANK1&quot;     &quot;ANKRD18A&quot; &quot;ANO3&quot;     &quot;ATP2C2&quot;   &quot;B4GALNT1&quot;
 [13] &quot;C1QL3&quot;    &quot;CCK&quot;      &quot;CCNA1&quot;    &quot;CCNB2&quot;    &quot;CD36&quot;     &quot;CDH1&quot;    
 [19] &quot;CDH12&quot;    &quot;CDK1&quot;     &quot;CEMIP&quot;    &quot;CHODL&quot;    &quot;CNTNAP3B&quot; &quot;COL25A1&quot; 
 [25] &quot;CRHBP&quot;    &quot;CRYM&quot;     &quot;CUX2&quot;     &quot;CXCL14&quot;   &quot;DDR2&quot;     &quot;ELOVL2&quot;  
 [31] &quot;FASLG&quot;    &quot;FGFR3&quot;    &quot;FILIP1&quot;   &quot;FSTL4&quot;    &quot;GAD1&quot;     &quot;GAD2&quot;    
 [37] &quot;GAS2L3&quot;   &quot;GJA1&quot;     &quot;HES1&quot;     &quot;HS3ST2&quot;   &quot;HTR2A&quot;    &quot;HTR2C&quot;   
 [43] &quot;ITGA8&quot;    &quot;ITGAM&quot;    &quot;KCNH5&quot;    &quot;KIT&quot;      &quot;LAMP5&quot;    &quot;LHX6&quot;    
 [49] &quot;LRRK1&quot;    &quot;LRRK2&quot;    &quot;LYPD6&quot;    &quot;LYPD6B&quot;   &quot;LYVE1&quot;    &quot;MEPE&quot;    
 [55] &quot;MYO5B&quot;    &quot;NCSTN&quot;    &quot;NDST4&quot;    &quot;NOTCH1&quot;   &quot;NPFFR2&quot;   &quot;NPNT&quot;    
 [61] &quot;NPY1R&quot;    &quot;NTNG1&quot;    &quot;NTNG2&quot;    &quot;NWD2&quot;     &quot;NXPH2&quot;    &quot;OTOGL&quot;   
 [67] &quot;P2RY12&quot;   &quot;P2RY13&quot;   &quot;PAX6&quot;     &quot;PCNA&quot;     &quot;PCSK1&quot;    &quot;PDGFD&quot;   
 [73] &quot;PHLDB2&quot;   &quot;PLCH1&quot;    &quot;PLD5&quot;     &quot;POU6F2&quot;   &quot;PSEN2&quot;    &quot;PTCHD4&quot;  
 [79] &quot;PVALB&quot;    &quot;RELN&quot;     &quot;RIT2&quot;     &quot;RORB&quot;     &quot;ROS1&quot;     &quot;RSPO2&quot;   
 [85] &quot;RXFP1&quot;    &quot;SAMD5&quot;    &quot;SERPINA3&quot; &quot;SFRP2&quot;    &quot;SLC17A6&quot;  &quot;SLC17A7&quot; 
 [91] &quot;SLC26A4&quot;  &quot;SNCG&quot;     &quot;SORCS1&quot;   &quot;SOX11&quot;    &quot;SPHKAP&quot;   &quot;SPON1&quot;   
 [97] &quot;SST&quot;      &quot;SULF1&quot;    &quot;SYNPR&quot;    &quot;TAC1&quot;     &quot;TACR1&quot;    &quot;TESPA1&quot;  
[103] &quot;THSD4&quot;    &quot;THSD7B&quot;   &quot;TOP2A&quot;    &quot;TP53&quot;     &quot;TPH2&quot;     &quot;TRAC&quot;    
[109] &quot;TRHDE&quot;    &quot;TRIL&quot;     &quot;TRPC5&quot;    &quot;TRPC6&quot;    &quot;VIP&quot;      &quot;VWC2L&quot;   
[115] &quot;WIF1&quot;     &quot;ZBBX&quot;     &quot;ZDHHC23&quot; </code></pre>
<pre class="r"><code># Annotations for genes not expressed in scRNAseq
brain %&gt;% filter(!Gene %in% unlist(exp_list)) %&gt;%
  select(`Annotation (Xenium)`) %&gt;%
  table() %&gt;%
  sort() %&gt;%
  as.data.frame()</code></pre>
<pre><code>           Annotation..Xenium. Freq
1           Glioblastoma (TME)    1
2                        Lamp5    1
3                          Vip    1
4                  Endothelial    2
5                        L5 IT    2
6                        L6 IT    2
7              Oligodendrocyte    2
8                          OPC    2
9                      L2/3 IT    3
10                       L6 CT    3
11                         L6b    3
12                   Astrocyte    4
13                       L5 ET    4
14                        Sncg    4
15                         Sst    4
16                  Chandelier    5
17                       L4 IT    6
18                     L5/6 NP    6
19                  Lamp5 Lhx6    6
20               Microglia-PVM    6
21               Proliferation    6
22                       Pvalb    6
23                        VLMC    6
24 Glioblastoma (Cancer cells)    8
25                  L6 IT Car3    8
26                        Pax6    8
27                   Sst Chodl    8</code></pre>
<pre class="r"><code># Total # of panel features expressed in scRNAseq
length(intersect(panel_gene_list[[&quot;brain&quot;]], unique(unlist(exp_list))))</code></pre>
<pre><code>[1] 149</code></pre>
<pre class="r"><code># % out of the whole panel expressed in scRNAseq
length(intersect(panel_gene_list[[&quot;brain&quot;]], unique(unlist(exp_list))))/length(panel_gene_list[[&quot;brain&quot;]])*100</code></pre>
<pre><code>[1] 56.01504</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ComplexHeatmap_2.18.0 RColorBrewer_1.1-3    viridis_0.6.3        
 [4] viridisLite_0.4.2     ggthemes_4.2.4        UpSetR_1.4.0         
 [7] googlesheets4_1.1.0   workflowr_1.7.1       ggrepel_0.9.3        
[10] ggpubr_0.6.0          lubridate_1.9.2       forcats_1.0.0        
[13] stringr_1.5.0         dplyr_1.1.2           purrr_1.0.1          
[16] readr_2.1.4           tidyr_1.3.0           tibble_3.2.1         
[19] ggplot2_3.4.2         tidyverse_2.0.0       SeuratDisk_0.0.0.9021
[22] Seurat_5.0.1          SeuratObject_5.0.1    sp_1.6-1             
[25] cli_3.6.1            

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.20       splines_4.3.0          later_1.3.1           
  [4] cellranger_1.1.0       polyclip_1.10-4        fastDummies_1.7.3     
  [7] lifecycle_1.0.3        rstatix_0.7.2          doParallel_1.0.17     
 [10] rprojroot_2.0.3        globals_0.16.2         processx_3.8.1        
 [13] lattice_0.21-8         hdf5r_1.3.8            MASS_7.3-60           
 [16] backports_1.4.1        magrittr_2.0.3         plotly_4.10.2         
 [19] sass_0.4.6             rmarkdown_2.22         jquerylib_0.1.4       
 [22] yaml_2.3.7             httpuv_1.6.11          sctransform_0.4.1     
 [25] spam_2.9-1             spatstat.sparse_3.0-1  reticulate_1.29       
 [28] cowplot_1.1.1          pbapply_1.7-0          abind_1.4-5           
 [31] Rtsne_0.16             BiocGenerics_0.48.1    git2r_0.32.0          
 [34] circlize_0.4.15        S4Vectors_0.40.2       IRanges_2.36.0        
 [37] irlba_2.3.5.1          listenv_0.9.0          spatstat.utils_3.0-3  
 [40] goftest_1.2-3          RSpectra_0.16-1        spatstat.random_3.1-5 
 [43] fitdistrplus_1.1-11    parallelly_1.36.0      leiden_0.4.3          
 [46] codetools_0.2-19       shape_1.4.6            tidyselect_1.2.0      
 [49] farver_2.1.1           stats4_4.3.0           matrixStats_1.0.0     
 [52] spatstat.explore_3.2-1 googledrive_2.1.0      jsonlite_1.8.5        
 [55] GetoptLong_1.0.5       ellipsis_0.3.2         progressr_0.13.0      
 [58] iterators_1.0.14       ggridges_0.5.4         survival_3.5-5        
 [61] foreach_1.5.2          tools_4.3.0            ica_1.0-3             
 [64] Rcpp_1.0.10            glue_1.6.2             gridExtra_2.3         
 [67] xfun_0.39              withr_2.5.0            fastmap_1.1.1         
 [70] fansi_1.0.4            callr_3.7.3            digest_0.6.31         
 [73] timechange_0.2.0       R6_2.5.1               mime_0.12             
 [76] colorspace_2.1-0       scattermore_1.2        tensor_1.5            
 [79] spatstat.data_3.0-1    utf8_1.2.3             generics_0.1.3        
 [82] data.table_1.14.8      httr_1.4.6             htmlwidgets_1.6.2     
 [85] whisker_0.4.1          uwot_0.1.14            pkgconfig_2.0.3       
 [88] gtable_0.3.3           lmtest_0.9-40          htmltools_0.5.5       
 [91] carData_3.0-5          dotCall64_1.0-2        clue_0.3-64           
 [94] scales_1.2.1           png_0.1-8              knitr_1.43            
 [97] rstudioapi_0.14        rjson_0.2.21           tzdb_0.4.0            
[100] reshape2_1.4.4         nlme_3.1-162           curl_5.0.0            
[103] GlobalOptions_0.1.2    cachem_1.0.8           zoo_1.8-12            
[106] KernSmooth_2.23-21     parallel_4.3.0         miniUI_0.1.1.1        
[109] pillar_1.9.0           vctrs_0.6.2            RANN_2.6.1            
[112] promises_1.2.0.1       car_3.1-2              xtable_1.8-4          
[115] cluster_2.1.4          evaluate_0.21          compiler_4.3.0        
[118] rlang_1.1.1            crayon_1.5.2           future.apply_1.11.0   
[121] ggsignif_0.6.4         labeling_0.4.2         ps_1.7.5              
[124] getPass_0.2-4          plyr_1.8.8             fs_1.6.2              
[127] stringi_1.7.12         deldir_1.0-9           munsell_0.5.0         
[130] lazyeval_0.2.2         spatstat.geom_3.2-1    Matrix_1.6-5          
[133] RcppHNSW_0.5.0         hms_1.1.3              patchwork_1.1.2       
[136] bit64_4.0.5            future_1.32.0          shiny_1.7.4           
[139] highr_0.10             ROCR_1.0-11            gargle_1.4.0          
[142] igraph_1.4.3           broom_1.0.4            bslib_0.4.2           
[145] bit_4.0.5             </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
